### DISCUSSION TITLE
Updatable image control from network

### DISCUSSION THREAD

**Discussion by noau** [woelper/oculante#308] [Ideas]
Hello there,

I'm so glad that this powerful image viewer supports viewing image passed from network, and I'm wondering whether remote control and update of image has/could be implemented, just like the protocol provided by [tev](https://github.com/Tom94/tev), which is unfortunately badly documented.
And here's my idea modified from tev's:

- **OpenImage**: Opens an image from a specified path on the machine it's running on, or passing the whole image data as what oculante is doing currently.
- **CreateImage**: Creates a blank image with a specified name, size, and set of channels. If an image with the specified name already exists, it would be overwritten. It would be beneficial to specify the color format here.
- **UpdateImage**: Updates the pixels in a rectangular region, or updates specific pixels defined by a (location, pixel data) pair.
- **CloseImage**: Closes a specified image.
- **ReloadImage**: Reloads an image from a specified path on the machine it's running on. This could also partially implement the feature discussed in [Reacting to file changes?](https://github.com/woelper/oculante/discussions/241).

**Comment by woelper** [comment]
Hey!
Sorry, I am really bad with keeping up with discussions. This is definitely nothing personal, I just don't always see the notifications.

OpenImage: This more or less could be done with network mode I guess? I would say there could be improvements, such as supporting all images, including animated ones.

CreateImage: I think I don't understand the use case.

CloseImage: I am not sure I understand why this is useful.

ReloadImage should not be necessary - that should work out of the box.


Given the above, I think it is very interesting to have some kind of "network protocol" that sends commands. Oculante also has operators, which could be enabled over the network - but I'd love to understand more what the problem is you are trying to solve in order to find the best solution to it.
